zerver/lib/string_validation.py

```diff
@@ -0,0 +1,37 @@
+import unicodedata
+
+from django.utils.translation import gettext as _
+
+from zerver.lib.exceptions import JsonableError
+from zerver.models import Stream
+
+unicode_non_chars = [
+    chr(x)
+    for x in list(range(0xFDD0, 0xFDF0))  # FDD0 through FDEF, inclusive
+    + list(range(0xFFFE, 0x110000, 0x10000))  # 0xFFFE, 0x1FFFE, ... 0x10FFFE inclusive
+    + list(range(0xFFFF, 0x110000, 0x10000))  # 0xFFFF, 0x1FFFF, ... 0x10FFFF inclusive
+]
+
+
+def check_invalid_unicode_character(var: str) -> int:
```

Comment:
It looks to me like the function should  return `bool` rather than `int`, since it's communicating one of two possible values.

To make the function name clearer, should it perhaps be `check_string_for_disallowed_characters` or something like that? Perhaps Alex will have a comment on that, given the discussion about the term "invalid X" earlier.